src/coreclr/jit/objectalloc.cpp

```diff
@@ -644,7 +733,8 @@ bool ObjectAllocator::CanLclVarEscapeViaParentStack(ArrayStack<GenTree*>* parent
                     // 2. Protect objects as interior (GCPROTECT_BEGININTERIOR() instead of GCPROTECT_BEGIN()).
                     // 3. Don't check that the object is in the heap in ValidateInner.
 
-                    canLclVarEscapeViaParentStack = true;
+                    canLclVarEscapeViaParentStack =
+                        !Compiler::s_helperCallProperties.IsNoEscape(comp->eeGetHelperNum(asCall->gtCallMethHnd));
```

Comment:
Is this still helpful?

Reply:
Yes, we need to know that CORINFO_HELP_UNBOX doesn't cause escape. Then later we rewrite it to a version that doesn't access the box at all.